src/test/java/org/opensearch/jobscheduler/utils/JobDetailsServiceIT.java

```diff
@@ -137,4 +142,57 @@ public void testDeleteNonExistingJobDetails() throws ExecutionException, Interru
         }, exception -> fail(exception.getMessage())));
     }
 
+    public void testUpdateIndexToJobDetails() throws ExecutionException, InterruptedException, TimeoutException {
+        String extensionId = "sample-extension";
+        CompletableFuture<Boolean> inProgressFuture = new CompletableFuture<>();
+        JobDetailsService jobDetailsService = new JobDetailsService(client(), this.clusterService, this.indicesToListen);
+
+        // Index Job Index name, actions, for extensionID
+        jobDetailsService.processJobDetailsForExtensionId(
+            "sample-job-index",
+            null,
+            "sample-job-parameter",
+            "sample-job-runner",
+            extensionId,
+            JobDetailsService.JobDetailsRequestType.JOB_INDEX,
+            ActionListener.wrap(jobDetailsWithoutJobType -> {
+                // Index Job Type
+                jobDetailsService.processJobDetailsForExtensionId(
+                    null,
+                    "sample-job-type",
+                    null,
+                    null,
+                    extensionId,
+                    JobDetailsService.JobDetailsRequestType.JOB_TYPE,
+                    ActionListener.wrap(jobDetails -> {
+                        // Ensure job details entry is valid
+                        assertEquals("sample-job-index", jobDetails.getJobIndex());
+                        assertEquals("sample-job-type", jobDetails.getJobType());
+                        assertEquals("sample-job-parameter", jobDetails.getJobParameterAction());
+                        assertEquals("sample-job-runner", jobDetails.getJobRunnerAction());
+
+                        // We'll have to invoke updateIndexToJobDetails as jobDetailsService is added as an indexOperationListener
+                        // onIndexModule
+                        jobDetailsService.updateIndexToJobDetails(extensionId, jobDetails);
+
+                        // Ensure indicesToListen is updated
+                        assertTrue(this.indicesToListen.contains(jobDetails.getJobIndex()));
+
+                        // Ensure indexToJobDetails is updated
+                        JobDetails entry = jobDetailsService.getIndexToJobDetails().get(extensionId);
+                        assertEquals(jobDetails.getJobIndex(), entry.getJobIndex());
```

Comment:
(Readability) It's not clear what expected/actual are here since the `jobDetails` itself is "actual" up at linen 169.  Can the expected here just be those same four strings?